client/hosting/performance/hooks/usePerformanceReport.ts

```diff
@@ -0,0 +1,110 @@
+import { useState, useCallback, useEffect } from 'react';
+import { useUrlBasicMetricsQuery } from 'calypso/data/site-profiler/use-url-basic-metrics-query';
+import { useUrlPerformanceInsightsQuery } from 'calypso/data/site-profiler/use-url-performance-insights';
+import { TabType } from 'calypso/performance-profiler/components/header';
+
+function isValidURL( url: string ) {
```

Comment:
Can we try using the `URL` class? It can be instantiated with the URL and will throw an error in case it's invalid.

```suggestion
function isValidURL( url: string ) {
    try {
        new URL(url);
        return true;
    } catch {
        return false;
    }
```

Something like that

Reply:
sounds good!